### DISCUSSION TITLE
Avoiding overwriting user's changes on file upload

### DISCUSSION THREAD

**Discussion by javierllorente** [WeblateOrg/weblate#7453] [Q&A]
Hello!

I am developing a Weblate library (jwl) which gets/submits an entire file. This means that a user could overwrite another user's changes, if the file he or she is translating has become outdated. From my experience there is usually only one translator assigned to a translation (file), but still.
Is there a way to get around this? i.e: something like POST /api/[...]/file/lock/

Cheers,
Javier

**Comment by nijel** [comment]
You can lock translations at the component level.

**Reply by javierllorente** [reply]
Wouldn't that lock all languages for that component?

**Reply by nijel** [reply]
Yes, it would.

**Reply by javierllorente** [reply]
That could make other translators not very happy ;(
It would be nice to be able to lock a file. Candidate for enhancement?

**Reply by nijel** [reply]
It used to be there (slightly differently implemented though), but it has proven to be problematic.

The translation level locking needs to open up locking to more people (to make it useful) and with more people, it is more likely that somebody forgets to unlock it after he is done.

What would be a more viable approach is https://github.com/WeblateOrg/weblate/issues/265, but still that would allow assigned translators to overwrite each other work.

**Reply by javierllorente** [reply]
> The translation level locking needs to open up locking to more people (to make it useful) and with more people, it is more likely that somebody forgets to unlock it after he is done.

A team coordinator should be able to unlock it. We could also have automatic unlocking after some time of inactivity.
Having translation assignments (#265) would be also a nice feature. The risk of overwriting other user's translations is minimised as well.
